Automattic/jetpack
# Boost: make the setting modules work more consistently

The modules of the Boost settings page all work in slightly different ways. This PR update them to fix that.
Any module with a meta form (Page Cache, the concatenation modules, Image quality) will use the CollapsibleMeta module to handle the expanded section.

Concatenate modules:
Instead of a cancel button, it uses the same show hide link other modules use, with up/down chevrons. ~~I think I'd rather leave the text of the link, instead of changing it to "Hide" as the cursor has to be moved if it was not in the right place.
If the text input form is closed, the value of the form is reset to the saved value now. It would retain an edited version before, unlike other modules.~~
The text of the expand link won't change now.
The Cancel button is removed and the form does not hide when it is saved.
Press enter or return to submit the setting, as well as clicking on Save.

Fixes #36274
Fixes #40252

## Proposed changes:

* Wrap the SCSS in a wrapper selector and adjust CSS to work.
* Use an up/down chevron to show and hide the meta form.
* Show/hide link uses `isExpanded` state like other modules now.
* Remove the Cancel button
* Add a description and help text.
* Remove `inputLabel` parameter from `MetaComponent` as it uses `datasyncKey` to decide if JS or CSS.
* Lots more, this is a big one.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
* Go to settings page and open and close the modules
* Edit the settings and save them.
* Reload the page to confirm the changes stick.
